[Filebeat][New Module] Mysql Enterprise Audit log

P1llus opened the pull request:
## What does this PR do?

This creates a module that focuses on logs and files created by MySQL. Currently this only has support for Enterprise Audit logs but can be extended in the future.

## Why is it important?

Adding support for more OOTB solutions for filebeat.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.


Field references: https://dev.mysql.com/doc/refman/8.0/en/audit-log-file-formats.html

elasticmachine commented:
Pinging @elastic/integrations-services (Team:Services)

elasticmachine commented:
Pinging @elastic/security-external-integrations (Team:Security-External Integrations)

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > There is a new build on-going so the previous on-going builds have been aborted.
  


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22273/detail/PR-22273/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22273/detail/PR-22273/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22273/detail/PR-22273/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-22273/detail/PR-22273/1//artifacts) 

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #22273 opened]`

* Reason: `Aborted from #2`

* Start Time: 2020-10-29T16:50:57.687+0000

* Duration: 51 min 28 sec


* Commit: 15ad7913555459e852070f350c090e8676d2a37c



#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 4475  |
| Skipped      | 561 |
| Total        | 5036   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->




<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2020-10-29T17:41:12.708Z] 3.4: Pulling from library/alpine
[2020-10-29T17:41:12.971Z] c1e54eec4b57: Pulling fs layer
[2020-10-29T17:41:13.234Z] c1e54eec4b57: Verifying Checksum
[2020-10-29T17:41:13.234Z] c1e54eec4b57: Download complete
[2020-10-29T17:41:13.497Z] c1e54eec4b57: Pull complete
[2020-10-29T17:41:13.497Z] Digest: sha256:b733d4a32c4da6a00a84df2ca32791bb03df95400243648d8c539e7b4cce329c
[2020-10-29T17:41:13.497Z] Status: Downloaded newer image for alpine:3.4
[2020-10-29T17:41:15.740Z] + python .ci/scripts/pre_archive_test.py
[2020-10-29T17:41:17.648Z] Copy ./x-pack/filebeat/build into build/x-pack/filebeat/build
[2020-10-29T17:41:17.672Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats/build
[2020-10-29T17:41:18.063Z] + rm -rf ve
[2020-10-29T17:41:18.063Z] + find . -type d -name vendor -exec rm -r {} ;
[2020-10-29T17:41:18.107Z] Recording test results
[2020-10-29T17:41:19.690Z] Stashed 4 file(s)
[2020-10-29T17:41:20.155Z] Client: Docker Engine - Community
[2020-10-29T17:41:20.155Z]  Version:           19.03.13
[2020-10-29T17:41:20.155Z]  API version:       1.40
[2020-10-29T17:41:20.155Z]  Go version:        go1.13.15
[2020-10-29T17:41:20.155Z]  Git commit:        4484c46d9d
[2020-10-29T17:41:20.155Z]  Built:             Wed Sep 16 17:02:36 2020
[2020-10-29T17:41:20.155Z]  OS/Arch:           linux/amd64
[2020-10-29T17:41:20.155Z]  Experimental:      false
[2020-10-29T17:41:20.155Z] 
[2020-10-29T17:41:20.155Z] Server: Docker Engine - Community
[2020-10-29T17:41:20.155Z]  Engine:
[2020-10-29T17:41:20.155Z]   Version:          19.03.13
[2020-10-29T17:41:20.155Z]   API version:      1.40 (minimum version 1.12)
[2020-10-29T17:41:20.155Z]   Go version:       go1.13.15
[2020-10-29T17:41:20.155Z]   Git commit:       4484c46d9d
[2020-10-29T17:41:20.155Z]   Built:            Wed Sep 16 17:01:06 2020
[2020-10-29T17:41:20.155Z]   OS/Arch:          linux/amd64
[2020-10-29T17:41:20.155Z]   Experimental:     false
[2020-10-29T17:41:20.155Z]  containerd:
[2020-10-29T17:41:20.155Z]   Version:          1.3.7
[2020-10-29T17:41:20.155Z]   GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
[2020-10-29T17:41:20.155Z]  runc:
[2020-10-29T17:41:20.155Z]   Version:          1.0.0-rc10
[2020-10-29T17:41:20.155Z]   GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
[2020-10-29T17:41:20.155Z]  docker-init:
[2020-10-29T17:41:20.155Z]   Version:          0.18.0
[2020-10-29T17:41:20.155Z]   GitCommit:        fec3683
[2020-10-29T17:41:36.195Z] [INFO] unstashV2: JOB_GCS_BUCKET is set. bucket param got precedency instead.
[2020-10-29T17:41:36.220Z] [INFO] unstashV2: JOB_GCS_CREDENTIALS is set. credentialsId param got precedency instead.
[2020-10-29T17:41:36.272Z] [Google Cloud Storage Plugin] Found 1 files to download from pattern: gs://beats-ci-temp/Beats/beats/PR-22273-1/source/source.tgz
[2020-10-29T17:41:36.296Z] [Google Cloud Storage Plugin] Downloading: Beats/beats/PR-22273-1/source/source.tgz to local path: /var/lib/jenkins/workspace/Beats_beats_PR-22273/source.tgz
[2020-10-29T17:41:50.660Z] + tar --version
[2020-10-29T17:41:51.069Z] + tar -xpf source.tgz
[2020-10-29T17:42:03.659Z] + rm source.tgz
[2020-10-29T17:42:04.353Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats
[2020-10-29T17:42:04.387Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats/filebeat-Lint-1603991247456
[2020-10-29T17:42:04.492Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats/x-pack-filebeat-Lint-1603991392587
[2020-10-29T17:42:04.583Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats/x-pack-filebeat-windows-windows-2019-1603991731081
[2020-10-29T17:42:04.673Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats/filebeat-windows-windows-2019-1603991855268
[2020-10-29T17:42:04.764Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats/filebeat-build-1603992373300
[2020-10-29T17:42:04.861Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats/x-pack-filebeat-build-1603993279315
[2020-10-29T17:42:05.282Z] + cat
[2020-10-29T17:42:05.282Z] + /usr/local/bin/runbld ./runbld-test-reports --job-name elastic+beats+pull-request
[2020-10-29T17:42:05.282Z] Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
[2020-10-29T17:42:11.870Z] runbld>>> runbld started
[2020-10-29T17:42:11.870Z] runbld>>> 1.6.12/f45d832f2ba0aa2722ab4ec1fda8ad140f027f8b
[2020-10-29T17:42:13.252Z] runbld>>> The following profiles matched the job 'elastic+beats+pull-request' in order of occurrence in the config (last value wins).
[2020-10-29T17:42:13.252Z] runbld>>> Matches in the system config:
[2020-10-29T17:42:13.252Z] runbld>>> - Matched ^elastic\+beats
[2020-10-29T17:42:13.252Z] runbld>>> - Matched ^elastic\+beats\+pull-request
[2020-10-29T17:42:14.646Z] runbld>>> Debug logging enabled.
[2020-10-29T17:42:14.646Z] runbld>>> Storing result
[2020-10-29T17:42:14.646Z] runbld>>> Store result: created {:total 2, :successful 2, :failed 0} 1
[2020-10-29T17:42:14.646Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1597739501209/t/20201029174214-066CAE0C
[2020-10-29T17:42:14.646Z] runbld>>> Adding system facts.
[2020-10-29T17:42:15.588Z] runbld>>> Adding vcs info for the latest commit:  4c4ef73220f3a063a700ecfdba3473677a38c1ca
[2020-10-29T17:42:15.588Z] runbld>>> >>>>>>>>>>>> SCRIPT EXECUTION BEGIN >>>>>>>>>>>>
[2020-10-29T17:42:15.849Z] runbld>>> Adding /usr/lib/jvm/java-8-openjdk-amd64/bin to the path.
[2020-10-29T17:42:15.849Z] Processing JUnit reports with runbld...
[2020-10-29T17:42:15.849Z] + echo 'Processing JUnit reports with runbld...'
[2020-10-29T17:42:16.109Z] runbld>>> <<<<<<<<<<<< SCRIPT EXECUTION END <<<<<<<<<<<<
[2020-10-29T17:42:16.109Z] runbld>>> DURATION: 28ms
[2020-10-29T17:42:16.109Z] runbld>>> STDOUT: 40 bytes
[2020-10-29T17:42:16.109Z] runbld>>> STDERR: 49 bytes
[2020-10-29T17:42:16.109Z] runbld>>> WRAPPED PROCESS: SUCCESS (0)
[2020-10-29T17:42:16.109Z] runbld>>> Searching for build metadata in /var/lib/jenkins/workspace/Beats_beats_PR-22273
[2020-10-29T17:42:17.053Z] runbld>>> Storing build metadata: 
[2020-10-29T17:42:17.053Z] runbld>>> Adding test report.
[2020-10-29T17:42:17.053Z] runbld>>> Searching for junit test output files with the pattern: TEST-.*\.xml$ in: /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats
[2020-10-29T17:42:17.992Z] runbld>>> Found 12 test output files
[2020-10-29T17:42:18.937Z] runbld>>> Test output logs contained: Errors: 0 Failures: 0 Tests: 5036 Skipped: 539
[2020-10-29T17:42:19.197Z] runbld>>> Storing result
[2020-10-29T17:42:19.197Z] runbld>>> FAILURES: 0
[2020-10-29T17:42:19.457Z] runbld>>> Store result: updated {:total 2, :successful 2, :failed 0} 2
[2020-10-29T17:42:19.457Z] runbld>>> BUILD: https://c150076387b5421f9154dfbf536e5c60.us-west1.gcp.cloud.es.io:9243/build-1597739501209/t/20201029174214-066CAE0C
[2020-10-29T17:42:19.457Z] runbld>>> Email notification disabled by environment variable.
[2020-10-29T17:42:19.457Z] runbld>>> Slack notification disabled by environment variable.
[2020-10-29T17:42:24.815Z] Running in /var/lib/jenkins/workspace/Beats_beats_PR-22273/src/github.com/elastic/beats
[2020-10-29T17:42:25.184Z] Running on Jenkins in /var/lib/jenkins/workspace/Beats_beats_PR-22273
[2020-10-29T17:42:25.310Z] [INFO] getVaultSecret: Getting secrets
[2020-10-29T17:42:25.389Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2020-10-29T17:42:26.283Z] + chmod 755 generate-build-data.sh
[2020-10-29T17:42:26.283Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22273/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22273/runs/1 ABORTED 3088325
[2020-10-29T17:42:26.284Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22273/runs/1/steps/?limit=10000 -o steps-info.json
[2020-10-29T17:42:27.194Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22273/runs/1/tests/?status=FAILED -o tests-errors.json
[2020-10-29T17:42:27.194Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/beats/PR-22273/runs/1/log/ -o pipeline-log.txt

```

</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
## :green_heart: Flaky test report
Tests succeeded.


<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

### Test stats :test_tube:
| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 4475  |
| Skipped      | 561 |
| Total        | 5036   |




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_flaky.id-->

leehinman commented on x-pack/filebeat/module/mysql/audit/_meta/fields.yml:
```diff
@@ -0,0 +1,121 @@
+    - name: login.user
+      type: keyword
+      description: >
+        A string representing the text of an SQL statement. The value can be empty. Long values may be truncated. The string, like the audit log file itself, is written using UTF-8 (up to 4 bytes per character), so the value may be the result of conversion.
```
I think this is the same description as the one from general_data.query.  Guessing it needs to be updated.

leehinman commented on x-pack/filebeat/module/mysql/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,971 @@
+        "mysql.audit.connection_id": 13,
+        "mysql.audit.id": 0,
+        "mysql.audit.login.user": "root",
+        "process.pid": "33038",
```
should be a long, not string. Probably need to convert instead of rename.

leehinman commented on x-pack/filebeat/module/mysql/audit/ingest/pipeline.yml:
```diff
@@ -0,0 +1,110 @@
+      handleMap(ctx);
+- rename:
+    field: mysql.audit.account.user
+    target_field: server.user.name
```
probably want to add server.user.name & client.user.name to related.user

leehinman commented on x-pack/filebeat/module/mysql/audit/ingest/pipeline.yml:
```diff
@@ -0,0 +1,110 @@
+    ignore_missing: true
+- rename:
+    field: mysql.audit.login.ip
+    target_field: client.ip
```
probably want to add to related.ip.  Also any value in doing geoip?

P1llus replied:
I think almost no connections that would create a audit entry would be a public IP, I do not mind adding it in, but I don't think it would resolve for 99% of the entries. WDYT?

leehinman replied:
I'm fine with skipping geoip.

leehinman reviewed (changes requested):
looking good.

andrewstucki commented on filebeat/docs/fields.asciidoc:
```diff
@@ -59,6 +59,7 @@ grouped in the following categories:
 * <<exported-fields-mongodb>>
 * <<exported-fields-mssql>>
 * <<exported-fields-mysql>>
+* <<exported-fields-mysql>>
```
You'll want to name this something like `exported-fields-mysql-enterprise`, it conflicts with the ID right above it.

andrewstucki commented on filebeat/docs/fields.asciidoc:
```diff
@@ -97685,6 +97686,257 @@ alias to: source.ip
 
 --
 
+[[exported-fields-mysql]]
```
Rename this anchor the same as above: `exported-fields-mysql-enterprise`

andrewstucki commented on filebeat/docs/fields.asciidoc:
```diff
@@ -97685,6 +97686,257 @@ alias to: source.ip
 --
 
+[[exported-fields-mysql]]
+== mysql fields
```
You'll want to specify that this is for the x-pack MySQL enterprise module versus the [OSS MySQL module](https://github.com/elastic/beats/tree/master/filebeat/module/mysql). Something like `MySQL Enterprise Fields`?

andrewstucki commented on filebeat/docs/fields.asciidoc:
```diff
@@ -97685,6 +97686,257 @@ alias to: source.ip
+[[exported-fields-mysql]]
+== mysql fields
+
+Some mysql module
```
😬

andrewstucki commented on x-pack/filebeat/modules.d/mysql.yml.disabled:
```diff
@@ -0,0 +1,13 @@
+# Module: mysql
+# Docs: https://www.elastic.co/guide/en/beats/filebeat/master/filebeat-module-mysql.html
+
+- module: mysql
```
I'm pretty sure that this is going to cause issues. Might need a module name change:

https://github.com/elastic/beats/blob/f23f05a33fb570f7100a80d9ae31ab720ad96d1d/filebeat/module/mysql/_meta/config.yml#L1

P1llus commented:
After discussion whether we want to merge this with the current OSS module we decided to rename the module into mysqlenterprise.

Once this is created into a package, both the OSS and the X-Pack module will merge into one.

P1llus commented:
Seems to be getting there now. @sayden anything you would want to add to this as well?

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "mysqlenterprise.audit.id": 0,
+        "mysqlenterprise.audit.startup_data.mysql_version": "8.0.22-commercial",
+        "mysqlenterprise.audit.startup_data.server_id": 1,
+        "process.args": [
```
what about `process.executable` and `process.command_line`?

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "mysqlenterprise.audit.class": "audit",
+        "mysqlenterprise.audit.connection_id": 0,
+        "mysqlenterprise.audit.id": 0,
+        "mysqlenterprise.audit.startup_data.mysql_version": "8.0.22-commercial",
```
`service.version`?

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "mysqlenterprise.audit.connection_id": 0,
+        "mysqlenterprise.audit.id": 0,
+        "mysqlenterprise.audit.startup_data.mysql_version": "8.0.22-commercial",
+        "mysqlenterprise.audit.startup_data.server_id": 1,
```
`service.ephemeral_id` or `service.id`?

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "input.type": "log",
+        "log.offset": 0,
+        "mysqlenterprise.audit.class": "audit",
+        "mysqlenterprise.audit.connection_id": 0,
```
do we know what it means if you have a `connection_id` field at startup? Just wondering if we should try and figure out an ECS mapping for it or not

P1llus replied:
I don't think there is any. At first I tried to find a "session ID" field we could use in ECS; but it seems that the session ID is only unique to the current active connections, so it will reset and use the same connection_id for another connection later according to its docs:

https://dev.mysql.com/doc/refman/8.0/en/information-functions.html#function_connection-id

Do you know any other field that might match? I tried to look in client, user and network but could not find anything that matches well

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "mysqlenterprise.audit.class": "connection",
+        "mysqlenterprise.audit.connection_data.connection_attributes._client_name": "libmysql",
+        "mysqlenterprise.audit.connection_data.connection_attributes._client_version": "8.0.22",
+        "mysqlenterprise.audit.connection_data.connection_attributes._os": "Linux",
```
should these map to `host.os` fields?

P1llus replied:
These should be in the `client` fields, `connection_data` is for client data and `startup_data` is for the host. However there was no `*.os` fields for `client` in the ECS definition :(

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "client.domain": "localhost",
+        "event.action": "mysql-connect",
+        "event.category": [
+            "database"
```
also `network`?

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "event.module": "mysqlenterprise",
+        "event.outcome": "success",
+        "event.type": [
+            "access",
```
also `connection`?

P1llus replied:
Was a bit unsure if access, was enough, adding in connection as well.

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "mysqlenterprise.audit.connection_id": 15,
+        "mysqlenterprise.audit.general_data.command": "Query",
+        "mysqlenterprise.audit.general_data.query": "CREATE USER 'audit_test_user'@'localhost' IDENTIFIED BY <secret>",
+        "mysqlenterprise.audit.general_data.sql_command": "create_user",
```
should this be categorized with `iam` values?

P1llus replied:
I think the issue with "sql_command" is that we would have to create a list of conditions and trying to catch all the different SQL commands it supports. 
I can try to add a few at least!

andrewstucki commented on x-pack/filebeat/module/mysqlenterprise/audit/test/mysql_audit_test.log-expected.json:
```diff
@@ -0,0 +1,1100 @@
+        "mysqlenterprise.audit.connection_id": 15,
+        "mysqlenterprise.audit.general_data.command": "Query",
+        "mysqlenterprise.audit.general_data.query": "GRANT ALL PRIVILEGES ON *.* TO 'audit_test_user2'@'hades.home'",
+        "mysqlenterprise.audit.general_data.sql_command": "grant",
```
same with these: `iam`?

P1llus commented:
@andrewstucki Added a better test log example for processes, added most of the different fields and left comments on the rest.
Also added some extra process fields in case we wanted to use them, please let me know what you think.

adriansr commented on filebeat/docs/modules/mysqlenterprise.asciidoc:
```diff
@@ -0,0 +1,80 @@
+
+This is a module for different types of MySQL logs. Currently focusing on data from the MySQL Enterprise Audit Plugin in JSON format.
+
+To configure the the Enterprise Audit Plugin to output in JSON format please follow the directions in the https://dev.mysql.com/doc/refman/8.0/en/audit-log-file-formats.html[MySQL Documentation]
```
Nit: Missing dot at the end.
```suggestion
To configure the the Enterprise Audit Plugin to output in JSON format please follow the directions in the https://dev.mysql.com/doc/refman/8.0/en/audit-log-file-formats.html[MySQL Documentation.]
```

adriansr commented on filebeat/docs/modules/mysqlenterprise.asciidoc:
```diff
@@ -0,0 +1,80 @@
+| login.ip                   | client.ip                      |
+| startup_data.os_version    | host.os.full                   |
+| startup_data.args          | process.args                   |
+| onnection_attributes._pid  | process.pid                    |
```
```suggestion
| connection_attributes._pid  | process.pid                    |
```

adriansr commented on filebeat/docs/modules/mysqlenterprise.asciidoc:
```diff
@@ -0,0 +1,80 @@
+MySQL Enterprise Audit fields are mapped to ECS in the following way:
+[options="header"]
+|==============================================================
+| Check Point Fields         | ECS Fields                     |
```
Is "Check Point" a concept in MySQL Audit logs or was this copied over from Check Point docs?

adriansr commented on x-pack/filebeat/module/mysqlenterprise/_meta/fields.yml:
```diff
@@ -0,0 +1,10 @@
+- key: mysqlenterprise
+  title: Mysql Enterprise
```
```suggestion
  title: MySQL Enterprise
```

adriansr commented on x-pack/filebeat/module/mysqlenterprise/_meta/fields.yml:
```diff
@@ -0,0 +1,10 @@
+- key: mysqlenterprise
+  title: Mysql Enterprise
+  description: >
+    Mysql Enterprise Audit module
```
Same
```suggestion
    MySQL Enterprise Audit module
```

adriansr commented on x-pack/filebeat/module/mysqlenterprise/audit/ingest/pipeline.yml:
```diff
@@ -0,0 +1,158 @@
+    lang: painless
+    if: ctx?.process?.args != null
+    source: |
+      ctx.process.args_count = ctx.process.args.length - 1;
```
According to ECS this should be the "length of the process.args array", not the length minus 1.

adriansr commented on x-pack/filebeat/module/mysqlenterprise/audit/ingest/pipeline.yml:
```diff
@@ -0,0 +1,158 @@
+    if: ctx?.process?.args != null
+    source: |
+      ctx.process.args_count = ctx.process.args.length - 1;
+      ctx.process.executable = ctx.process.args[0];
```
Being extra paranoid here, I think this will cause an exception when process.args is an empty list (non-null, but zero length).

adriansr commented on x-pack/filebeat/module/mysqlenterprise/_meta/config.yml:
```diff
@@ -0,0 +1,10 @@
+
+    # Set paths for the log files when file input is used.
+    # Should only be used together with file input
+    # var.paths: /home/user/mysqlauditlogs/audit.*.log
```
I *think* this won't work if the user uncomments the line, as `vars.paths` is expected to be a list not a string.

```suggestion
    # var.paths:
    #  - /home/user/mysqlauditlogs/audit.*.log
```

adriansr reviewed (dismissed):
All in all it's looking good, but I have a few little suggestions here in there.
